apps/src/gamelab/AnimationTab/PiskelEditor.jsx

```diff
@@ -65,6 +70,39 @@ const PiskelEditor = React.createClass({
     if (newProps.selectedAnimation !== this.props.selectedAnimation) {
       this.loadSelectedAnimation_(newProps);
     }
+    if (newProps.pendingAnimationAddition
+      && newProps.selectedAnimation === newProps.pendingAnimationAddition.key) {
+      this.addPendingAnimationAddition(newProps.pendingAnimationAddition);
+    }
+  },
+
+  addPendingAnimationAddition(animationProps) {
+    const key = this.props.selectedAnimation;
+    if (!animationProps) {
+      throw new Error('No props present for animation with key ' + key);
+    }
+
+    this.isLoadingAnimation_ = true;
+    if (animationProps.props.blankFrame) {
+      this.piskel.addBlankFrame();
+      this.isLoadingAnimation_ = false;
+      this.props.removePendingAnimationAddition();
+    } else if (animationProps.dataURI) {
+      this.piskel.loadAdditionalFrames(
+        animationProps.dataURI,
+        animationProps.props.frameSize.x,
+        animationProps.props.frameSize.y,
+        animationProps.props.frameDelay,
```

Comment:
`animationProps.props` Are we able to differentiate these different sets of props?